src/controllers/simplified-account/services/payment-links/create/payment-link-information.controller.ts

```diff
@@ -0,0 +1,194 @@
+    .optional({ checkFalsy: true })
+    .trim()
+    .isLength({ max: 255 })
+    .withMessage('Details must be 255 characters or fewer'),
+]
+
+const makeNiceURL = (string: string): string => {
+  return slugify(removeIndefiniteArticles(string))
+}
+
+const publicAuthClient = new PublicAuthClient()
+
+function get(req: ServiceRequest, res: ServiceResponse) {
+  const service = req.service
+  const { account } = req
+
+  const backLinkUrl = formatServiceAndAccountPathsFor(
+    paths.simplifiedAccount.paymentLinks.index,
+    service.externalId,
+    account.type
+  )
+
+  const friendlyURL = process.env.PRODUCTS_FRIENDLY_BASE_URI
+  const languageValue: string = getLanguageFromQuery(req)
+  const isWelsh: boolean = languageValue === supportedLanguage.WELSH
+  const language: SupportedLanguage = isWelsh ? supportedLanguage.WELSH : supportedLanguage.ENGLISH
+
+  const serviceName: string = (language === 'cy' && service.serviceName.cy)
+    ? String(service.serviceName.cy)
+    : String(service.serviceName.en)
+
+  return response(req, res, 'simplified-account/services/payment-links/create/index', {
+    service,
+    account,
+    backLink: backLinkUrl,
+    formValues: {},
+    friendlyURL,
+    serviceName,
+  })
+}
+
+async function post(req: ServiceRequest, res: ServiceResponse, next: NextFunction) {
+  const service = req.service
+  const { account } = req
+  const body: CreatePaymentLinkBody = req.body
+
+  await Promise.all(validations.map((validation) => validation.run(req)))
+  const errors = validationResult(req)
+
+  if (!errors.isEmpty()) {
+    const formattedErrors = formatValidationErrors(errors)
+    const backLinkUrl = formatServiceAndAccountPathsFor(
+      paths.simplifiedAccount.paymentLinks.index,
+      service.externalId,
+      account.type
+    )
+
+    const friendlyURL = process.env.PRODUCTS_FRIENDLY_BASE_URI
+    const languageValue: string = getLanguageFromQuery(req)
+    const isWelsh: boolean = languageValue === supportedLanguage.WELSH
+    const language: SupportedLanguage = isWelsh ? supportedLanguage.WELSH : supportedLanguage.ENGLISH
+
+    const serviceName: string = (language === 'cy' && service.serviceName.cy)
+      ? String(service.serviceName.cy)
+      : String(service.serviceName.en)
+
+    return response(req, res, 'simplified-account/services/payment-links/create/index', {
+      service,
+      account,
+      errors: {
+        summary: formattedErrors.errorSummary,
+        formErrors: formattedErrors.formErrors,
+      },
+      backLink: backLinkUrl,
+      formValues: body,
+      friendlyURL,
+      serviceName,
+    })
+  }
+
+  try {
+    const createTokenRequest = new CreateTokenRequest()
+      .withGatewayAccountId(account.id)
+      .withServiceExternalId(service.externalId)
+      .withServiceMode(account.type)
+      .withDescription(`Token for "${body.name ?? ''}" payment link`)
+      .withCreatedBy((req.user as { email: string }).email)
+      .withTokenUsageType('PRODUCTS')
```

Comment:
as above, this likely needs moving to the end of the journey so that we don't create tokens unnecessarily